src/main/java/org/owasp/wrongsecrets/challenges/ChallengeUI.java

```diff
@@ -1,19 +1,31 @@
 package org.owasp.wrongsecrets.challenges;
 
-import lombok.Getter;
-import org.owasp.wrongsecrets.RuntimeEnvironment;
-
 import java.util.Comparator;
 import java.util.List;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
+import lombok.Getter;
+import org.owasp.wrongsecrets.RuntimeEnvironment;
 
 /**
  * Wrapper class to move logic from Thymeleaf to keep logic in code instead of the html file.
  */
 @Getter
 public class ChallengeUI {
 
+    private record DifficultyUI(Difficulty difficulty) {
```

Comment:
this is very cool :D , but wouldn't using the difficulity its int make it easier? and then just have a static max dificulty set in the abstract challenge class?

Reply:
and can you please add javadoc :) ?